algebra/tests/vec/various.c

```diff
@@ -303,3 +310,103 @@ TEST_GROUP_RUNNER(group_vec_sub)
 	RUN_TEST_CASE(group_vec_sub, vec_sub_std);
 	RUN_TEST_CASE(group_vec_sub, vec_sub_biggerValues);
 }
+
+
+/* ##############################################################################
+ * ------------------------        vec_cross tests       ------------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_vec_cross);
+
+
+TEST_SETUP(group_vec_cross)
+{
+}
+
+
+TEST_TEAR_DOWN(group_vec_cross)
+{
+}
+
+
+TEST(group_vec_cross, vec_cross_std)
+{
+	vec_t A = V1;
+	vec_t B = V2;
+	vec_t C;
+
+	vec_cross(&A, &B, &C);
+
+	TEST_ASSERT_EQUAL_FLOAT(A.y * B.z - A.z * B.y, C.x);
+	TEST_ASSERT_EQUAL_FLOAT(A.z * B.x - A.x * B.z, C.y);
+	TEST_ASSERT_EQUAL_FLOAT(A.x * B.y - A.y * B.x, C.z);
+}
+
+
+TEST(group_vec_cross, vec_cross_biggerValues)
+{
+	vec_t A = V3;
+	vec_t B = V4;
+	vec_t C;
+
+	vec_cross(&A, &B, &C);
+
+	TEST_ASSERT_EQUAL_FLOAT(A.y * B.z - A.z * B.y, C.x);
+	TEST_ASSERT_EQUAL_FLOAT(A.z * B.x - A.x * B.z, C.y);
+	TEST_ASSERT_EQUAL_FLOAT(A.x * B.y - A.y * B.x, C.z);
+}
+
+
+TEST(group_vec_cross, vec_cross_perpendicular)
+{
+	vec_t A = V5;
+	vec_t B = V6;
+	vec_t C;
+
+	vec_cross(&A, &B, &C);
+
+	TEST_ASSERT_EQUAL_FLOAT(A.y * B.z - A.z * B.y, C.x);
+	TEST_ASSERT_EQUAL_FLOAT(A.z * B.x - A.x * B.z, C.y);
+	TEST_ASSERT_EQUAL_FLOAT(A.x * B.y - A.y * B.x, C.z);
+
+	vec_cross(&B, &A, &C);
+
+	TEST_ASSERT_EQUAL_FLOAT(B.y * A.z - B.z * A.y, C.x);
+	TEST_ASSERT_EQUAL_FLOAT(B.z * A.x - B.x * A.z, C.y);
+	TEST_ASSERT_EQUAL_FLOAT(B.x * A.y - B.y * A.x, C.z);
+}
+
+
+TEST(group_vec_cross, vec_cross_parallel)
+{
+	vec_t A = V2;
+	vec_t B = A;
+	vec_t C;
+
+	vec_times(&B, POS_SCALAR);
```

Comment:
This test do two things, better to pinpoint what happens now :)
```suggestion

	/* Parallel with common direction */
	vec_times(&B, POS_SCALAR);
```

Reply:
Done